pkg/appconsts/versioned_consts.go

```diff
@@ -94,7 +100,16 @@ func UpgradeHeightDelay(chainID string, v uint64) int64 {
 			return v3.UpgradeHeightDelay
 		}
 		return v2.UpgradeHeightDelay
-	default:
-		return v3.UpgradeHeightDelay
+	case v3.Version:
+		switch chainID {
+		case ArabicaChainID:
+			return numBlocksPerDay
+		case MochaChainID:
+			return numBlocksPerDay * 2
+		case MainnetChainID:
+			return v3.UpgradeHeightDelay
+		}
 	}
+	// TODO: this should panic because this should never be invoked for v4+
```

Comment:
[not blocking]
is there an issue for this TODO?

Reply:
No issue but it may not be worth including this TODO because it will live in the v3.x line which we won't really be updating b/c we're about to merge all the v4 stuff to `main` soon